docs: GOVERNANCE: Added quorum requirement for governance votes and clarified the role of subproject lead (#175)
argoproj/argoproj: community/membership.md

### How inactivity is measured

Inactive members are defined as members of one of the Argoproj Organizations
with no significant contributions within the last 12 months. Contributions are
measured using the CNCF [DevStats project], GitHub activity history, and other media
by which a history of contributions can be reliably determined.

If an actively contributing member is accidentally removed this way, they may open an
issue to quickly be re-instated. 

After an extended period away from the project with no activity,
removed members would need to re-familiarize themselves with the current state
before being able to contribute effectively. 

## Change in membership roles

A change in membership role may be approved by the appropriate subproject lead in consultation with other Argo Project leads and approvers. By convention, role changes are discussed in a meeting with Argo Project approvers before being finalized.

## Appointment of Subproject Leads

> Leads serve a unique and critical role in the direction and organization of each subproject. Each subproject will generally have two co-leads, who share responsibility. There is no fixed term for leads and they often serve until they retire or, in rare cases, are replaced by [voting](GOVERANCE.md). Leads are encouraged to give other qualified members of the community an opportunity to serve. Former leads are welcome to serve as leads again in the future.

Reviewer: ```suggestion
Leads serve a unique and critical role in the direction and organization of each subproject. Each subproject will generally have two co-leads, who share responsibility. There is no fixed term for leads and they often serve until they retire or, in rare cases, are replaced by [voting](GOVERNANCE.md). Leads are encouraged to give other qualified members of the community an opportunity to serve. Former leads are welcome to serve as leads again in the future.
```

When a new lead needs to be chosen for a subproject, the availability of the role must be communicated in advance to the Argo maintainer community. Any qualified maintainer may nominate themself for the role. The final decision for the appointment of a new lead will be made by consensus of all existing Argo Project leads and approvers.

Key factors in the selection of new subproject leads are:
- History of participation and leadership in the Argo project and user communities.
- Demonstrated high standards for personal and professional conduct.

> - Leads should reflect and represent the diversity of the Argo Community 

Reviewer: Could you clarify this?

Author: Diversity in ideas, backgrounds, gender, etc.

> - Sustained technical contributions to the subproject including driving significant new featuers.

Reviewer: ```suggestion
- Sustained technical contributions to the subproject including driving significant new features.
```

> - Sustained technical contributions to the subproject including driving significant new features.

Reviewer 2: Anyone who has "made a dent".

Reviewer 2: "Sufficiently demonstrated a strong technical understanding and commitment towards the project"

### How inactivity is measured

Inactive members are defined as members of one of the Argoproj Organizations
with no significant contributions within the last 12 months. Contributions are
measured using the CNCF [DevStats project], GitHub activity history, and other media
by which a history of contributions can be reliably determined.

If an actively contributing member is accidentally removed this way, they may open an
issue to quickly be re-instated. 

After an extended period away from the project with no activity,
removed members would need to re-familiarize themselves with the current state
before being able to contribute effectively. 

## Change in membership roles

A change in membership role may be approved by consensus of the Argo Project leads and approvers. By convention, role changes are discussed in a meeting with Argo Project approvers before being finalized.

## Appointment of Subproject Leads

> Leads serve a unique and critical role in the direction and organization of each subproject. Each subproject will generally have two co-leads, who share responsibility. There is no fixed term for leads and they often serve until they retire or, in rare cases, are replaced by [voting](GOVERNANCE.md). Leads are encouraged to give other qualified members of the community an opportunity to serve. Former leads are welcome to serve as leads again in the future.

Reviewer: > Each subproject will generally have two co-leads

Could you clarify whether there is a hard limit on the number of co-leads?

> Former leads are welcome to serve as leads again in the future

What would the process be?

Author: The same as for new leads

Reviewer: Thanks. What about the question on the number of co-leads?

> When a new lead needs to be chosen for a subproject, the availability of the role must be communicated in advance to the Argo maintainer community. Any qualified maintainer may nominate themself for the role. The final decision for the appointment of a new lead will be made by consensus of all existing Argo Project leads and approvers.

Reviewer: > consensus of all existing Argo Project leads and approvers

This "all" seems different from the quorum requirement that you added in the PR and the existing 2/3 supermajority rule? What if some approvers are inactive or don't vote?

Author: Will remove "all"

Key factors in the selection of new subproject leads are:
- History of participation and leadership in the Argo project and user communities.
- Demonstrated high standards for personal and professional conduct.

> - Leads should reflect and represent the diversity of the Argo Community 

Reviewer 3: ```suggestion
- Leads should reflect and represent the diversity of the Open Source Community 
```
